packages/components/Button/src/useButton.ts

```diff
@@ -8,31 +8,68 @@ import { isHighContrast } from '@fluentui-react-native/theming-utils';
 
 import type { ButtonProps, ButtonInfo } from './Button.types';
 
+// On win32 we don't want to fire the onClick event if the Button
+// hasn't received a key down event first. This prevents behavior
+// like the button firing after you tab to it white Enter is pressed
+// and then releasing Enter, or the Menu reopening since it closes
+// onKeyDown while the Button operates onKeyUp.
+const shouldOnlyFireIfPressed = Platform.OS === ('win32' as any);
+
 export const useButton = (props: ButtonProps): ButtonInfo => {
   const defaultComponentRef = React.useRef(null);
   const {
-    onClick,
     accessible,
     accessibilityRole,
     componentRef = defaultComponentRef,
     disabled,
+    onBlur,
+    onClick,
+    onKeyDown,
     loading,
     enableFocusRing,
     focusable,
     ...rest
   } = props;
 
   const isDisabled = !!disabled || !!loading;
+  const [isKeyPressed, setIsKeyPressed] = React.useState<boolean>(false);
+
   // GH #1336: Set focusRef to null if button is disabled to prevent getting keyboard focus.
   const focusRef = isDisabled ? null : componentRef;
   const onClickWithFocus = useOnPressWithFocus(focusRef, onClick);
-  const pressable = usePressableState({ ...rest, onPress: onClickWithFocus });
-  const onKeyUpProps = useKeyProps(onClick, ' ', 'Enter');
+  const onBlurInner = React.useCallback(
+    (e) => {
+      setIsKeyPressed(false);
+      onBlur?.(e);
+    },
+    [onBlur],
+  );
+  const pressable = usePressableState({ ...rest, onPress: onClickWithFocus, onBlur: shouldOnlyFireIfPressed ? onBlurInner : onBlur });
+
+  const onKeyDownInner = React.useCallback(
+    (e) => {
+      if ((!disabled && e.nativeEvent.key === 'Enter') || e.nativeEvent.key === ' ') {
+        setIsKeyPressed(true);
+      }
+      onKeyDown?.(e);
```

Comment:
should other keydowns invalidate the pressed state?

Reply:
Native doesn't seem to do that so I don't think so.